src/lib.rs

```diff
@@ -622,13 +630,54 @@ impl GenerateTasksService for GenerateTasksServiceImpl {
             for task in &build_variant.tasks {
                 if task.name == BURN_IN_TAGS {
                     if self.gen_burn_in {
-                        burn_in_tag_build_variant_names.extend(
-                            self.evg_config_utils
-                                .lookup_and_split_by_whitespace_build_variant_expansion(
-                                    BURN_IN_TAG_BUILD_VARIANTS,
-                                    build_variant,
-                                ),
-                        );
+                        let burn_in_tag_build_variants = self
+                            .evg_config_utils
+                            .lookup_and_split_by_whitespace_build_variant_expansion(
+                                BURN_IN_TAG_BUILD_VARIANTS,
+                                build_variant,
+                            );
```

Comment:
Can we also check that `burn_in_tag_build_variants` vector is not empty? That should mean that `BURN_IN_TAG_BUILD_VARIANTS` expansion is absent or empty.

And sorry, I just remembered, but we also want to check that those values from `BURN_IN_TAG_BUILD_VARIANTS` expansion exist in `build_variant_map`. We should also error out in this case.